src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java

```diff
@@ -409,6 +505,77 @@ private int buildsOnExecutor(Task task, Executor exec) {
     }
 
     /**
+     * Get the count of currently executing {@link PlaceholderTask}s on a given {@link Executor} for a given {@link Run}
+     * and list of {@link FlowNode}s in that run that have been throttled.
+     *
+     * @param run The {@link Run} we care about.
+     * @param exec The {@link Executor} we're checking on.
+     * @param flowNodes The list of {@link FlowNode}s associated with that run that have been throttled with a particular
+     *                  category.
+     * @return 1 if there's something currently executing on that executor and it's of that run and one of the provided
+     * flow nodes, 0 otherwise.
+     */
+    private int pipelinesOnExecutor(@Nonnull Run<?,?> run, @Nonnull Executor exec, @Nonnull List<FlowNode> flowNodes) {
+        final Queue.Executable currentExecutable = exec.getCurrentExecutable();
+        if (currentExecutable != null) {
+            SubTask parent = currentExecutable.getParent();
+            if (parent instanceof PlaceholderTask) {
+                PlaceholderTask task = (PlaceholderTask)parent;
+                if (run.equals(task.run())) {
+                    if (isTaskThrottledPipeline(task, flowNodes)) {
+                        return 1;
+                    }
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    private boolean isTaskThrottledPipeline(Task origTask, List<FlowNode> flowNodes) {
+        if (origTask instanceof PlaceholderTask) {
+            PlaceholderTask task = (PlaceholderTask)origTask;
+            try {
+                FlowNode firstThrottle = firstThrottleStartNode(task.getNode());
```

Comment:
This really deserves a proper API so you can get rid of the `workflow-durable-task-step` dependency. It was never intended as an API plugin.